// @antv/g2plot/lib/plots/stock/constant.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DEFAULT_OPTIONS = exports.DEFAULT_TOOLTIP_OPTIONS = exports.TREND_COLOR = exports.TREND_DOWN = exports.TREND_UP = exports.TREND_FIELD = exports.Y_FIELD = void 0;
var plot_1 = require("../../core/plot");
var utils_1 = require("../../utils");
exports.Y_FIELD = '$$stock-range$$';
exports.TREND_FIELD = 'trend';
exports.TREND_UP = 'up';
exports.TREND_DOWN = 'down';
exports.TREND_COLOR = ['#ef5350', '#26a69a'];
/** tooltip 配置 */
exports.DEFAULT_TOOLTIP_OPTIONS = {
    showTitle: false,
    showMarkers: false,
    showCrosshairs: true,
    shared: true,
    crosshairs: {
        type: 'xy',
        follow: true,
    },
    itemTpl: '<li class="g2-tooltip-list-item" data-index={index}>' +
        '<span style="background-color:{color};" class="g2-tooltip-marker"></span>' +
        '{name}{value}</li>',
};
/**
 * 散点图 默认配置项
 */
exports.DEFAULT_OPTIONS = utils_1.deepAssign({}, plot_1.Plot.getDefaultOptions(), {
    // 设置默认图表 tooltips
    tooltip: exports.DEFAULT_TOOLTIP_OPTIONS,
    interactions: [{ type: 'tooltip' }],
    legend: {
        position: 'top-left',
    },
});
