// @ant-design/charts/lib/graphs/contants.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.defaultLabelCfg = exports.defaultTitleLabelCfg = exports.Margin = exports.defaultNodeAnchorPoints = exports.defaultLayout = exports.defaultEdgeArrowStyle = exports.defaultNodeStyle = exports.defaultNodeSize = exports.defaultStateStyles = void 0;
var g6_1 = __importDefault(require("@antv/g6"));
// 默认交互状态
exports.defaultStateStyles = {
    hover: {
        stroke: '#1890ff',
        lineWidth: 2,
    },
};
// card 默认节点大小
exports.defaultNodeSize = [120, 40];
// 默认节点样式
exports.defaultNodeStyle = {
    stroke: '#40a9ff',
};
// 默认箭头样式
exports.defaultEdgeArrowStyle = {
    endArrow: {
        path: g6_1.default.Arrow.vee(10, 10),
        fill: '#ccc',
    },
};
// card 默认布局系数
exports.defaultLayout = {
    type: 'compactBox',
    direction: 'LR',
    getId: function (d) {
        return d.id;
    },
    getHeight: function () {
        return 60;
    },
    getWidth: function () {
        return 16;
    },
    getVGap: function () {
        return 16;
    },
    getHGap: function () {
        return 100;
    },
};
// 默认 anchor 连接点
exports.defaultNodeAnchorPoints = [
    [0, 0.5],
    [1, 0.5],
];
// card 内部 padding | margin
exports.Margin = 8;
// card title 默认样式
exports.defaultTitleLabelCfg = {
    fill: '#fff',
    fontSize: 12,
};
// card body|footer  默认样式
exports.defaultLabelCfg = {
    fill: '#000',
    fontSize: 12,
};
