// @ant-design/charts/es/graphs/contants.js
import G6 from '@antv/g6';
// 默认交互状态
export var defaultStateStyles = {
    hover: {
        stroke: '#1890ff',
        lineWidth: 2,
    },
};
// card 默认节点大小
export var defaultNodeSize = [120, 40];
// 默认节点样式
export var defaultNodeStyle = {
    stroke: '#40a9ff',
};
// 默认箭头样式
export var defaultEdgeArrowStyle = {
    endArrow: {
        path: G6.Arrow.vee(10, 10),
        fill: '#ccc',
    },
};
// card 默认布局系数
export var defaultLayout = {
    type: 'compactBox',
    direction: 'LR',
    getId: function (d) {
        return d.id;
    },
    getHeight: function () {
        return 60;
    },
    getWidth: function () {
        return 16;
    },
    getVGap: function () {
        return 16;
    },
    getHGap: function () {
        return 100;
    },
};
// 默认 anchor 连接点
export var defaultNodeAnchorPoints = [
    [0, 0.5],
    [1, 0.5],
];
// card 内部 padding | margin
export var Margin = 8;
// card title 默认样式
export var defaultTitleLabelCfg = {
    fill: '#fff',
    fontSize: 12,
};
// card body|footer  默认样式
export var defaultLabelCfg = {
    fill: '#000',
    fontSize: 12,
};
