// @ant-design/charts/es/graphs/indented.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React, { useEffect } from 'react';
import G6 from '@antv/g6';
import { ErrorBoundary } from '../base';
import useGraph from '../hooks/useGraph';
import ChartLoading from '../util/createLoading';
import { getGraphSize, getGraphId } from './utils';
import { bindDefaultEvents, renderGraph } from './utils';
import { registerNodes } from './customItems';
import { defaultNodeAnchorPoints, defaultNodeSize, defaultStateStyles, defaultLayout, defaultNodeStyle, defaultEdgeArrowStyle, } from './contants';
var graphs = {};
registerNodes();
var IndentedTreeGraph = function (props) {
    var data = props.data, className = props.className, style = props.style, width = props.width, height = props.height, _a = props.nodeType, nodeType = _a === void 0 ? 'card' : _a, _b = props.edgeType, edgeType = _b === void 0 ? 'cubic-horizontal' : _b, _c = props.behaviors, behaviors = _c === void 0 ? ['zoom-canvas', 'drag-canvas'] : _c, _d = props.nodeAnchorPoints, nodeAnchorPoints = _d === void 0 ? defaultNodeAnchorPoints : _d, _e = props.nodeSize, nodeSize = _e === void 0 ? defaultNodeSize : _e, layout = props.layout, nodeStyle = props.nodeStyle, edgeStyle = props.edgeStyle, nodeStateStyles = props.nodeStateStyles, edgeStateStyles = props.edgeStateStyles, _f = props.collapseExpand, collapseExpand = _f === void 0 ? true : _f, titleStyle = props.titleStyle, bodyStyle = props.bodyStyle, footerStyle = props.footerStyle, footerValueStyle = props.footerValueStyle, _g = props.showArrow, showArrow = _g === void 0 ? true : _g, onReady = props.onReady, loading = props.loading, loadingTemplate = props.loadingTemplate, errorTemplate = props.errorTemplate;
    var container = React.useRef(null);
    var graph = React.useRef(null);
    var graphId = getGraphId(graph);
    useGraph(graphs[graphId], props, container);
    useEffect(function () {
        var graphSize = getGraphSize(width, height, container);
        var graph = graphs[graphId];
        if (!graph) {
            graph = new G6.TreeGraph({
                container: container.current,
                width: graphSize[0],
                height: graphSize[1],
                modes: {
                    default: behaviors,
                },
                defaultNode: {
                    type: nodeType,
                    size: nodeSize,
                    anchorPoints: nodeAnchorPoints,
                    titleStyle: titleStyle,
                    bodyStyle: bodyStyle,
                    footerStyle: footerStyle,
                    footerValueStyle: footerValueStyle,
                },
                defaultEdge: {
                    type: edgeType,
                },
                nodeStateStyles: __assign(__assign({}, defaultStateStyles), nodeStateStyles),
                edgeStateStyles: __assign(__assign({}, defaultStateStyles), edgeStateStyles),
                layout: __assign(__assign({}, defaultLayout), layout),
            });
            graphs[graphId] = graph;
        }
        graph.node(function (node) {
            if (typeof nodeStyle === 'function') {
                return {
                    style: nodeStyle(node, graph),
                };
            }
            return {
                style: __assign(__assign({}, defaultNodeStyle), nodeStyle),
            };
        });
        graph.edge(function (edge) {
            if (typeof edgeStyle === 'function') {
                return {
                    style: edgeStyle(edge, graph),
                };
            }
            return {
                style: __assign(__assign({ stroke: '#ccc' }, (showArrow && defaultEdgeArrowStyle)), edgeStyle),
            };
        });
        if (collapseExpand) {
            bindDefaultEvents(graph, collapseExpand);
        }
        renderGraph(graph, data);
        if (onReady) {
            onReady(graph);
        }
        return function () {
            if (graphs[graphId]) {
                graphs[graphId].destroy();
                delete graphs[graphId];
            }
        };
    }, []);
    return (React.createElement(ErrorBoundary, { errorTemplate: errorTemplate },
        loading && React.createElement(ChartLoading, { loadingTemplate: loadingTemplate }),
        React.createElement("div", { className: className, style: style, ref: container })));
};
export default IndentedTreeGraph;
